pycbc/transforms.py

```diff
@@ -612,7 +612,101 @@ def inverse_jacobian(self, maps):
+        self.m1 = m1
+        self.m2 = m2
+        self.f_lower = f_lower
+        self.phi_ref = phi_ref
+        self.newIota = newIota
+        self.news1 = news1
+        self.news2 = news2
+        self._inputs = [self.iota, self.s1, self.s2, self.m1, self.m2, self.f_lower, self.phi_ref]
+        self._outputs = [self.newIota, self.news1, self.news2]
+        super(alignTotalSpin, self).__init__()   ############################################## what??????
 
+    def transform(self, maps):
+        """
+        Rigidly rotate binary so that the total angular momentum has the given
+        inclination (iota) instead of the orbital angular momentum. Return
+        the new inclination, s1, and s2. s1 and s2 are dimensionless spin.
+        Note: the spins are assumed to be given in the frame defined by the orbital
+        angular momentum.  
+        """
+        
+        iota = self.iota
+        s1 = self.s1
+        s2 = self.s2
+        m1 = self.m1
+        m2 = self.m2
+        f_lower = self.f_lower
+        phi_ref = self.phi_ref
+
+        # Calculate the quantities required by jframe_to_l0frame
+        if np.linalg.norm(maps[s1][:2]) != 0 or np.linalg.norm(maps[s1][:2]) != 0:
+            thetaJN = maps[iota]
+            phiJL = np.pi
+            l = np.array([0,0,1.])
+            theta1 = np.arccos(np.dot(maps[s1], l)/np.linalg.norm(maps[s1]))
+            theta2 = np.arccos(np.dot(maps[s2], l)/np.linalg.norm(maps[s2]))
+            phi12 = np.arccos(np.dot(maps[s1][:2], maps[s2][:2]) / np.linalg.norm(maps[s1][:2]) / \
+                              np.linalg.norm(maps[s2][:2]))
+            
+            #print(maps[s1])
+
+            news1 = np.zeros(3)
+            news2 = np.zeros(3)
+            
+            output = \
+            jframe_to_l0frame(maps[m1], maps[m2], maps[f_lower], maps[phi_ref], thetaJN, phiJL,
+                      np.linalg.norm(maps[s1]), np.linalg.norm(maps[s2]),
+                      theta1, theta2,
+                      phi12)
+            
+            newIota = output['inclination']      #not elegant, just quickly done for testing
+            news1[0] = output['spin1x']
+            news1[1] = output['spin1y']
+            news1[2] = output['spin1z']
+            news2[0] = output['spin2x']
+            news2[1] = output['spin2y']
+            news2[2] = output['spin2z']
+            
+            #print(output['spin1x'])   ### WIP!!!!
+            
+            out = {self.newIota: newIota, self.news1: news1, self.news2: news2}
+            return self.format_output(maps, out)
+        else:
+            out = {self.newIota: iota, self.news1: s1, self.news2: s2}
+            return self.format_output(maps, out)
+        
+######################################################################################################### TEST ################################################################################################################################################
```

Comment:
Remove this as well

Reply:
Hi, thank you for the feedback, and thanks for making me aware of issues I should have resolved before doing the PR. I'll take some time today to have a look over these and hopefully get this in a much better state.